alert-common/src/main/java/com/synopsys/integration/alert/common/workflow/processor/SummaryMessageContentProcessor.java

```diff
@@ -105,60 +109,78 @@ private ProviderMessageContent summarize(final ProviderMessageContent message) {
-            try {
-                if (!summarizedLinkableItems.isEmpty()) {
-                    ComponentItem newComponentItem = messageOperationCombiner.createNewComponentItem(componentItem, summarizedLinkableItems);
-                    summarizedCategoryItems.add(newComponentItem);
-                }
-            } catch (AlertException e) {
-                // If this happens, it means there is a bug in the Collector logic.
-                throw new AlertRuntimeException(e);
-            }
+    private LinkedHashSet<ComponentItem> summarizeCategoryComponentCounts(Set<ComponentItem> componentItems) {
+        Map<String, Map<ComponentItemPriority, Set<ComponentItem>>> itemsByCategory = new LinkedHashMap<>();
+
+        for (final ComponentItem item : componentItems) {
+            Map<ComponentItemPriority, Set<ComponentItem>> itemsByPriority = itemsByCategory.computeIfAbsent(item.getCategory(), ignored -> new LinkedHashMap<>());
+            Set<ComponentItem> itemList = itemsByPriority.computeIfAbsent(item.getPriority(), ignored -> new LinkedHashSet<>());
+            itemList.add(item);
         }
 
-        return summarizedCategoryItems;
-    }
+        LinkedHashSet<ComponentItem> componentSummaries = new LinkedHashSet<>();
+        for (Map.Entry<String, Map<ComponentItemPriority, Set<ComponentItem>>> entry : itemsByCategory.entrySet()) {
+            for (Map.Entry<ComponentItemPriority, Set<ComponentItem>> priorityEntries : entry.getValue().entrySet()) {
+                try {
+                    Set<ComponentItem> items = priorityEntries.getValue();
 
-    private Map<String, ComponentAttributeMap> collectComponentItemData(Set<ComponentItem> componentItemsForOperation) {
-        Map<String, ComponentAttributeMap> itemsByCategory = new LinkedHashMap<>();
-        for (final ComponentItem categoryItem : componentItemsForOperation) {
-            ComponentAttributeMap itemsOfSameName = categoryItem.getItemsOfSameName();
-            String summaryKey = createSummaryKey(categoryItem.getComponentKeys(), itemsOfSameName);
+                    List<LinkableItem> attributeSet = items.stream()
+                                                          .map(ComponentItem::getComponentAttributes)
+                                                          .flatMap(Set::stream)
+                                                          .filter(LinkableItem::isSummarizable)
+                                                          .collect(Collectors.toList());
+                    if (!attributeSet.isEmpty()) {
+                        Set<Long> notificationIds = items.stream()
+                                                        .map(ComponentItem::getNotificationIds)
+                                                        .flatMap(Set::stream)
+                                                        .collect(Collectors.toSet());
+                        Optional<ComponentItem> componentItem = items.stream().findFirst();
```

Comment:
Since we only use this variable once, we could immediately map this to Operation and get the value orElse some default (UPDATE?).